frappe/monitor.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies Pvt. Ltd. and Contributors
+# MIT License. See license.txt
+
+from __future__ import unicode_literals
+
+from datetime import datetime
+import json
+import traceback
+import frappe
+import os
+import uuid
+
+
+MONITOR_REDIS_KEY = "monitor-transactions"
+
+
+def start(transaction_type="request", method=None, kwargs=None):
+	if frappe.conf.monitor:
+		frappe.local.monitor = Monitor(
+			transaction_type=transaction_type, method=method, kwargs=kwargs
+		)
+
+
+def stop(response=None):
+	if frappe.conf.monitor and hasattr(frappe.local, "monitor"):
+		frappe.local.monitor.dump(response)
+
+
+def log_file():
+	return os.path.join(frappe.utils.get_bench_path(), "logs", "monitor.json.log")
+
+
+class Monitor:
+	def __init__(self, transaction_type=None, method=None, kwargs=None):
+		try:
+			self.site = frappe.local.site
+			self.timestamp = datetime.utcnow()
+			self.transaction_type = transaction_type
+			self.uuid = uuid.uuid4()
+
+			if self.transaction_type == "request":
+				self.data = frappe.form_dict
+				self.headers = dict(frappe.request.headers)
+				self.ip = frappe.local.request_ip
+				self.method = frappe.request.method
+				self.path = frappe.request.path
+			else:
+				self.kwargs = kwargs
+				self.method = method
+		except Exception:
+			traceback.print_exc()
+
+	def dump(self, response=None):
+		try:
+			timediff = datetime.utcnow() - self.timestamp
+			# Obtain duration in microseconds
+			self.duration = int(timediff.total_seconds() * 1000000)
+			data = {
+				"uuid": self.uuid,
+				"duration": self.duration,
+				"site": self.site,
+				"timestamp": self.timestamp.isoformat(sep=" "),
```

Comment:
```suggestion
				"timestamp": self.timestamp.isoformat(sep=str(" ")),
```
to avoid...
<img width="1440" alt="Screenshot 2020-02-24 at 3 59 57 PM" src="https://user-images.githubusercontent.com/13928957/75214442-993e9780-57b3-11ea-94f2-dbfb96739a81.png">